databricks/koalas/frame.py

```diff
@@ -6801,8 +6825,6 @@ def melt(self, id_vars=None, value_vars=None, var_name=None,
             value_vars = []
         elif isinstance(value_vars, str):
             value_vars = [(value_vars,)]
-        elif isinstance(value_vars, tuple):
-            value_vars = [value_vars]
```

Comment:
The root cause is not here.
We should check the specified vars are included in `column_index`.
E.g.,

```py
>>> df.melt(value_vars='Z')
Traceback (most recent call last):
...
pyspark.sql.utils.AnalysisException: "Can't extract value from pairs#15: need struct type but got string;"
```

This is necessary for multi-index columns support, but we might want to do the same thing for `id_vars` to follow pandas' behavior. I don't like the behavior, though. :)

Reply:
Thank you for the review. 👍 

I'll fix it in the other direction after comparing the melt function of pandas and koalas.